tests/testthat/test_as_version.R

```diff
@@ -0,0 +1,12 @@
+context("as_version")
+
+test_that("Bad version string", {
+  version_string_good <- "1.14.6"
+  version_string_bad  <- "Cairo: 1.14.6"
```

Comment:
I see. Isn't this already taken care of there: https://github.com/lionel-/vdiffr/blob/master/R/testthat-ui.R#L164

The CRAN failure was due to something else AFAICT btw.

Reply:
Yes, I think that did take care of it.  

Side note, I just found `base::package_version`, which does exactly the same work as `as_version`, but also handles both `.` and `-` as separators and notifies about parsing errors:

```
> base::package_version("1.14.6")
[1] ‘1.14.6’
> base::package_version("1.14-6")
[1] ‘1.14.6’
> base::package_version("Cairo: 1.2-3")
Error: invalid version specification ‘Cairo: 1.2-3’
``` 

Note that this function is hidden in `vdiffr` by `vdiffr::package_version` which calls `utils::packageVersion(pkg)`.  Perhaps it would be worthwhile renaming `vdiffr::package_version` to `vdiffr::package_version_str` or similar to avoid the name conflict.

In any case, I've got to get back to the work I get paid for.

Reply:
wow, a snake_case function in base R! Thanks for the tip, it looks like I should use that constructor.

Reply:
I think there are examples of *all* name formatting patterns in base R, 😁